cfme/automate/dialogs/dialog_element.py

```diff
@@ -0,0 +1,238 @@
+
+    def element_loc(self, element_type):
+        return self.browser.element('.//dialog-editor-field/div[@class="form-group"]'
+                                    '/label[normalize-space(.)={}]'.format(quote(element_type)))
+
+    def add_another_element(self, element):
+        """Method to add element."""
+        view = navigate_to(self, 'Edit')
+        dragged_element = element.get('element_information').get('choose_type')
+        box_label = element.get('element_information').get('ele_label')
+        view.dd.drag_and_drop(dragged_element, box_label)
+        view.element.edit_element(dragged_element)
+        view.fill(element.get('element_information'))
+        if element.get('options') is not None:
+            view.Options.fill(element.get('options'))
+        view.ele_save_button.click()
+        view.save_button.click()
+        view = self.create_view(DetailsDialogView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+
+    def reorder_elements(self, add_element, second_element, element_data):
+        """Method to add element and interchange element positions.
+           This method updates a dialog and adds a second element.The position
+           of two elements are then interchanged to test for error.(BZ-1238721)
+
+        Args:
+            add_element - flag if second element needs to be added.
+            second_element - The second element to be added to the dialog.
+            element_data - Already existing first element's data.
+        """
+        view = navigate_to(self, 'Edit')
+        if add_element:
+            dragged_element = second_element.get('element_information').get('choose_type')
+            box_label = second_element.get('element_information').get('ele_label')
+            view.dd.drag_and_drop(dragged_element, box_label)
+            view.element.edit_element(dragged_element)
+            view.fill(second_element.get('element_information'))
+            view.ele_save_button.click()
+
+        dragged_element = element_data.get('element_information').get("ele_label")
+        dropped_element = second_element.get('element_information').get("ele_label")
+        self.browser.drag_and_drop(self.element_loc(dragged_element),
+                                   self.element_loc(dropped_element))
+        view.save_button.click()
+        view = self.create_view(DetailsDialogView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+
+
+@attr.s
+class ElementCollection(BaseCollection):
+
+    ENTITY = Element
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def create(self, element_data):
+        for element in element_data:
+            view = navigate_to(self, "Add")
+            dragged_element = element.get('element_information').get('choose_type')
+            view.dd.drag_and_drop(dragged_element, self.parent.box_label)
+            view.fill(element.get('element_information'))
```

Comment:
Given how often this pattern is repeated I'd certainly like a better way to implement it - since Options is a tab nested view under element_information, is the separate fill call even necessary? if a nested fill dictionary is passed the first time, it should fill the tab as well.  That or just calling it without checking None first as I suggested above.